src/Features/Core/Portable/DocumentationComments/CodeFixes/AbstractAddDocCommentNodesCodeFixProvider.cs

```diff
@@ -0,0 +1,166 @@
+                        if (indexOfNode > 0 && paramNodeSiblings[indexOfNode - 1] is TXmlTextSyntax)
+                        {
+                            nodeAfterNewParamNode = paramNodeSiblings[indexOfNode - 1];
+                        }
+                    }
+                }
+                
+                var newNodeList = new SyntaxNode[]
+                {
+                    GetNewNode(parameterName, isFirstNodeInComment: newDocComment.ChildNodes().First() == nodeAfterNewParamNode)
+                };
+
+                newDocComment = nodeBeforeNewParamNode != null
+                    ? newDocComment.InsertNodesAfter(nodeBeforeNewParamNode, newNodeList)
+                    : newDocComment.InsertNodesBefore(nodeAfterNewParamNode, newNodeList);
+            }
+
+            var newRoot = root.ReplaceNode(docCommentNode, newDocComment.WithAdditionalAnnotations(Formatter.Annotation));
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        private IEnumerable<TXmlElementSyntax> GetNodes(SyntaxNode docComment)
+        {
+            var paramNodes = docComment.ChildNodes().OfType<TXmlElementSyntax>()
+                                                    .Where(w => GetXmlElementLocalName(w) == NodeName);
+
+            // Prefer to place the doc comment in the outer level, as in auto-created comments
+            if (!paramNodes.Any())
+            {
+                paramNodes = docComment.DescendantNodes(descendIntoChildren: _ => true)
+                    .OfType<TXmlElementSyntax>()
+                    .Where(w => GetXmlElementLocalName(w) == NodeName);
+            }
+
+            return paramNodes;
+        }
+
+        private bool NodeExists(IEnumerable<TXmlElementSyntax> paramNodes, string name)
+        {
+            return paramNodes.Select(GetNameAttributes)
+                             .Where(nameAttributes => nameAttributes.Count == 1)
+                             .Any(nameAttributes => nameAttributes.Select(GetValueFromNameAttribute).Contains(name));
+        }
+
+        protected TXmlElementSyntax GetLastParamNodeCorrespondingToParamInList(IEnumerable<TXmlElementSyntax> paramNodeList, List<string> methodParamSubset)
+        {
+            TXmlElementSyntax nodeAfterNewParamNode = null;
+
+            foreach (var paramNode in paramNodeList)
+            {
+                var paramNameForNode = GetNameAttributes(paramNode);
+
+                // param node is missing `name` attribute or there are multiple `name` attributes
+                if (paramNameForNode.Count != 1)
+                {
+                    continue;
+                }
+                
+                if (methodParamSubset.Contains(GetValueFromNameAttribute(paramNameForNode.Single())))
+                {
+                    nodeAfterNewParamNode = paramNode;
+                }
+            }
+
+            return nodeAfterNewParamNode;
```

Comment:
So i'm feeling like a lot of this code is pretty complex and difficult to grok.  The simple algorithm i was thinking should exist would be the following:

```
For each parameter you need to add
  If the parameter is the first parameter in the parameter list
    Try to find an existing param-node.  If you find it, add the new param-node to the start of it.
    Otherwise, add the param-node to the end of the doc-comment (or maybe after the summary if it exists.)
  If hte parameter is not the first parameter in the parameter list
    find the param-node corresponding** to the previous parameter in the param list and add your node after that.
```

** Note: there must be that corresponding param-node by induction.  i.e. you've ensured it exists through previous iterations of the loop.